src/service/allocator/allocator.service.ts

```diff
@@ -0,0 +1,372 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { PrismaService } from '../../db/prisma.service';
+import {
+  getAllocatorBiggestClientDistribution,
+  getAllocatorRetrievability,
+} from '../../../prisma/generated/client/sql';
+import { HistogramHelper } from '../../helper/histogram.helper';
+import { RetrievabilityWeekResponseDto } from '../../types/retrievabilityWeekResponse.dto';
+import { groupBy } from 'lodash';
+import { ProviderComplianceScoreRange } from '../../types/providerComplianceScoreRange';
+import { SpsComplianceWeekResponseDto } from '../../types/spsComplianceWeekResponse.dto';
+import { SpsComplianceWeekDto } from '../../types/spsComplianceWeek.dto';
+import { differenceInCalendarDays } from 'date-fns';
+
+@Injectable()
+export class AllocatorService {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly histogramHelper: HistogramHelper,
+  ) {}
+
+  async getAllocatorRetrievability() {
+    const allocatorWeeks = await this.prismaService.allocators_weekly.findMany({
+      select: {
+        week: true,
+      },
+      orderBy: {
+        week: 'desc',
+      },
+      where: {
+        week: {
+          lte: new Date(),
+        },
+      },
+      skip: 0,
+      take: 2,
+    });
+
+    const lastStoredFullWeek = allocatorWeeks.find(
+      (p) => differenceInCalendarDays(new Date(), p.week) >= 7,
+    );
```

Comment:
1. Mixing `luxon` and `date-fns` in one project looks like a mess - lets standardize on single lib
2. How about just directly calculating which week we should get? like `where: { week: DateTime.now().minus({ week: 1 }).startOf('week') }`

Reply:
fixed